### ISSUE TITLE
Content Security Policy issues when trying firevim with an Monaco based online IDE

### ISSUE THREAD

**Issue by ThijmenStar** [glacambre/firenvim#1315]
- OS Version: Pop!_OS 21.10 x86_64 (linux kernel 5.15.15-76051515-generic)
- Browser Version:  Firefox 96.0
- Browser Addon Version: 0.2.12
- Neovim Plugin Version: 0.2.12 

### What I tried to do
My university (TU Delft) uses an online IDE for most of it's (computer science) courses called [weblab](https://weblab.tudelft.nl/). I would really like to use vim shortcuts in this online editor. 

I did some digging and found out that under the hood they are using Monaco. I figured this extension should work and it did for the Monaco demo but unfortunately not for weblab's version.

### What happened
Instead the editor flashes black for a fraction of a second and then returns to it's original state. This is what the js console is saying:
```
This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”.
[index.html](moz-extension://20838920-61a2-4482-bf02-540a3e9d307e/index.html)
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). 3 moz-extension
Error: unwrap(...).monaco is undefined undefined
Error: unwrap(...).monaco is undefined [input.ts:187:16](webpack://Firenvim/src/input.ts)
Error: unwrap(...).monaco is undefined undefined
```

It seems it has something to do with the sites [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP). Is there a way to bypass this somehow?

**Comment by ThijmenStar** [comment]
Might be the same issue as #1311 though I'm not sure it's caused by the same thing

**Comment by glacambre** [comment]
Hi, thanks for the report. I think you're right, this sounds like a CSP issue. If I figure out how to solve #1193 this problem might go away :).

**Comment by albertas** [comment]
> Might be the same issue as #1311 though I'm not sure it's caused by the same thing

Yes, I confirm, that the cause of #1311 is the same. I get `Content Security Policy: The page's settings blocked the loading of a resource at inline ("script-src").` error when trying to start Firenvim in AWS console.

**Comment by JohnFinn** [comment]
Would it be possible to implement a workaround by modifying site's javascript? I created a small [firefox extension](https://github.com/JohnFinn/web_vim_remap) that reconfigures monaco-based editors on a couple of sites. Does anybody know if it would be possible to come up with such code injection to js, so that firevim starts working in such cases?
